lms/djangoapps/certificates/generation_handler.py

```diff
@@ -164,6 +165,11 @@ def _can_generate_allowlist_certificate(user, course_key):
         log.info(f'{user.id} : {course_key} does not have an enrollment. Certificate cannot be generated.')
         return False
 
+    if not CourseMode.is_eligible_for_certificate(enrollment_mode):
```

Comment:
Should we be accessing this model/model method directly across Django apps? (I fully realize I wasn't doing a great job asking this question in previous reviews but this is something I'm trying to keep in the forefront of my mind moving forward.)

Reply:
I wondered about that. There isn't any existing API for it (everything calls the model directly today if they need this method). I wasn't sure if we wanted to add API as we went, or leave that to the app's owners. 

Either way works for me.